### ISSUE TITLE
Feature: Unify admin and email passwords

### ISSUE THREAD

**Issue by naturalethic** [mjl-/mox#90]
I'm working on a client to present a unified account management, email, and admin.  I know you have a plan to introduce a auth tokens, so you could consider a unified auth by (username or email) + password.  The token, or an api call could reveal permissions.

**Comment by mjl-** [comment]
A new mechanism would work the same or the admin account and regular email accounts.
Or are you suggesting we should unify the accounts? E.g. have an account that has the special admin/operator privileges? 

On the one hand, it's perhaps a simpler mental model and admins of single-person setups won't have to deal with two passwords. On the other hand, it may be neater to have these roles completely separate.

I'm hoping to merge the current account (non-admin) page into the webmail pages. The admin pages will probably stay separate, so there will be a distinction. Though we could share a token between webmail and admin pages.

In the future, we may also want a new role: domain owner. Currently you're either the operator/admin of all domains, or an end-user with only access to your own account. A role like domain owner would give some of the functionality currently only available to an admin/operator, like creating new accounts with an address on a domain. We don't have to do anything with this idea now (implementing it would be quite involved), but good to keep in mind when designing a new scheme.

**Comment by naturalethic** [comment]
In the UI I'm working on, the server admin, account admin, and eventually webmail, are all part of the same front end.  So it's a bit goofy UX wise to have to log in twice to what is visibly the same interface.

But beyond that, I think architecturally there is a lot of flexibility in a unified user/role/permissions schema, where you can introduce finer grained authorizations as you bring in more features.

Personally, I would implement this using a basic JWT auth (login -> token pack with user and authorization info), and a schema that could initially just be user+role(s) (just admin/mail user for now), later user+role+permission (to customize and add roles with fine-grained permissions).  I wouldn't go so far as to implement OAuth.  Keep it in-house, custom, and tight.

**Comment by gedw99** [comment]
Can I add my 2 cents .., 

passkey support . I js e been on a streak to convert everything to use passkeys so do vr this is relevant I want to raise it .

mig you a simple to use example checkout Pocketbase . It allows admin to toggle what auth is exposed to uses . Pretty slick ..

https://github.com/pocketbase/pocketbase

so gme golang import 
Automatic SQLite db
Golang or js api ..

**Comment by gedw99** [comment]
Agree. Login to the web demo and go into admin auth . You can toggle all the ways users can authentication.

**Comment by gedw99** [comment]
It’s a one kind golang import to get that. I think it’s a huge leap up for mox . Theirs is a golang api to tap into it too .
